lightning/src/ln/channelmanager.rs

```diff
@@ -4431,8 +4424,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> Writeable f
 	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
 		let _consistency_lock = self.total_consistency_lock.write().unwrap();
 
-		writer.write_all(&[SERIALIZATION_VERSION; 1])?;
-		writer.write_all(&[MIN_SERIALIZATION_VERSION; 1])?;
+		write_ver_prefix!(writer, SERIALIZATION_VERSION, MIN_SERIALIZATION_VERSION);
```

Comment:
Route and NetworkGraph: yea, good catch, added.
EnforcingSigner is currently test-only (and may very well never be exposed as lightning-signer basically does what we'd want it to be), but I added a comment about it,
Features is a on-the-wire BOLT struct so we can't differ from the prescribed serialization format there.

Reply:
Question: so for keysend, I think I'm going to be adding/modifying fields in `PendingHTLCInfo `. Would this mean that versioning needs to be added to this struct (I believe this field is persisted in `Channel` so it could have potential compatibility issues)? Or does it mean I'll need to bump the serialization version in Channel? I guess I'm not sure of the exact procedure for changing struct fields going forward.

Reply:
Right, so for enums its easy - we usually have 0-N already defined and we can just add N+1... For something like `PendingHTLCInfo` we could maybe pass a `version` in to the `read` method, which would imply defining a new `VersionedReadable` trait. Of course that isn't exactly ideal because it means that even people who don't use keysend can't read new `ChannelManager`s with an old LDK version. If we want to be really broad about supporting old versions reading LDK structs from newer versions that don't use optional features, we should probably move the entire serialization of everything into TLVs (but in a separate PR).